src/ITensorNetworks/boundary_mps.jl

```diff
@@ -359,27 +372,12 @@ function sqnorm_approx(ψ::Matrix{ITensor}; center, cutoff, maxdim)
   ψ′ = addtags(linkinds, ψ, "ket")
   # TODO: implement contract(commoninds, ψ′, ψᴴ)
   tn = ψ′ .* ψᴴ
-
-  # Contract in every direction
-  combiner_gauge = combiners(linkinds, tn)
-  tnᶜ = insert_gauge(tn, combiner_gauge)
-  boundary_mpsᶜ = contract_approx(tnᶜ; maxdim=maxdim, cutoff=cutoff)
-
-  tn_cacheᶜ = contraction_cache(tnᶜ, boundary_mpsᶜ)
-  tn_cache = insert_gauge(tn_cacheᶜ, combiner_gauge)
-  _boundary_mps = boundary_mps(tn_cache)
-
-  #
   # Insert projectors horizontally (to measure e.g. properties
   # in a row of the network)
-  #
-
-  tn_projected = insert_projectors(tn, _boundary_mps; center=center)
+  tn_projected = insert_projectors(tn; center=center, cutoff=cutoff, maxdim=maxdim)
```

Comment:
Part of the reason why I had written it the previous way (precomputing the boundary MPS in a separate step) was because in general we will reuse the boundary MPS to create many different set of projectors for different terms in our cost function, such as the squared norm network and different Hamiltonian terms with interactions in different directions in our lattice (horizontal and vertical).

For now this is ok just to get this working, but we will want to reorganize the code so that first the boundary MPS are computed, then different sets of projectors are formed from those boundary MPS for different terms in our cost function (so that we aren't redoing work recomputing the same boundary MPS multiple times).

Reply:
I reorganized it a bit, splitting bmps out from the insert_projectors function